src/compiler/types.ts

```diff
@@ -688,6 +689,7 @@ namespace ts {
         Loop = 2,                           // Automatically generated identifier with a preference for '_i'.
         Unique = 3,                         // Unique name based on the 'text' property.
         Node = 4,                           // Unique name based on the node in the 'original' property.
+        OptimisticUnique = 5,               // Unique name based on the 'text' property, first instance won't use '_#' if there's no conflict

```

Comment:
YES. But is there any case where we don't want to just use `OptimisticUnique`? Maybe we should make this the behavior of `Unique` after this PR.

Reply:
@aozgaa has a reasonable point that you potentially want the `Unique` behavior in some cases (not here) to indicate that the output was machine-generated to some extent.